jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -894,11 +894,53 @@ void java_object_factoryt::gen_nondet_struct_init(
       code_function_callt{init_func->symbol_expr(), {address_of_exprt{expr}}});
 }
 
+/// Generate code block that verifies that an expression of type float or
+/// double has integral value. For example, for a float expression floatVal we
+/// generate:
+/// int assume_integral_tmp;
+/// assume_integral_tmp = NONDET(int);
+/// ASSUME FLOAT_TYPECAST(assume_integral_tmp, float, __CPROVER_rounding_mode)
+///   == floatVal
+/// \param expr The expression we want to make an assumption on
+/// \param type The type of the expression
+/// \param location Source location associated with the expression
+/// \param [out] symbol_table Symbol table, a new auxiliary symbol will be added
+/// \return Code block constructing the auxiliary nondet integer and an
+///   assume statement that the integer is equal to the value of the expression
+static code_blockt assume_expr_integral(
+  const exprt &expr,
+  const typet &type,
+  const source_locationt &location,
+  symbol_table_baset &symbol_table)
```

Comment:
:bulb: You might like to take a `allocate_local_symbolt` (in `nondet.h`) rather than a mutable symbol table to make the code more easily reusable in smart harness.

Reply:
Cool, thank you! Done

Reply:
I left this change in a separate commit for easier reviewing